crates/ra_hir/src/source_id.rs

```diff
@@ -132,6 +135,18 @@ impl AstIdMap {
         // trait does not change ids of top-level items, which helps caching.
         bfs(node, |it| {
             if let Some(module_item) = ast::ModuleItem::cast(it.clone()) {
+                // Collect attributes from `StructDef`s and `EnumDef`s
```

Comment:
I think if we store attributes as token trees in raw times, we wouldn't need to introduce ids for attributes, and this seems like a good thing to do

Reply:
Not sure what to do to resolve this.